src/vmm/src/lib.rs

```diff
@@ -361,11 +345,9 @@ impl VMM {
             eprintln!("Failed to set raw mode on terminal. Stdin will echo.");
         }
 
-        for mut vcpu in self.vcpus.drain(..) {
-            let _ = thread::Builder::new().spawn(move || loop {
-                vcpu.run();
-            });
-        }
+        // TODO: should we handle this in another way rather than a panic?
```

Comment:
Yep, I think so. The `vmm` crate here is a component in itself and it should be up to the app that consumes it to deal with a VMM that won't start. Maybe the error will be recoverable at some point and the UI could offer a retry possibility.